### ISSUE TITLE
[Bug] _preprocess_data not Implemented in base.py Estimator class

### ISSUE THREAD

**Issue by qchempku2017** [CederGroupHub/sparse-lm#5]
I've got this issue when trying to initialize an L2L0 class:
```
E               Traceback (most recent call last):
E                 File "C:\Users\66422\PycharmProjects\sparse-lm\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 686, in _fit_and_score
E                   estimator.fit(X_train, y_train, **fit_params)
E                 File "C:\Users\66422\PycharmProjects\sparse-lm\sparselm\model\base.py", line 78, in fit
E                   X, y, X_offset, y_offset, X_scale = self._preprocess_data(
E               AttributeError: 'L2L0' object has no attribute '_preprocess_data'
```

This might be that sklearn has changed it's reference to _preprocess_data from an object method in LinearModel to a global method called "preprocess_data_" in sklearn.linear_model._base. Do we have other functionalities that we wish to use in our implementation, such that we need to keep this function as an object method in the Estimator class? Otherwise we can just overwrite self._preprocessdata with the global method in sklearn.

**Comment by lbluque** [comment]
Thanks @qchempku2017 

What version of sklearn are you using?

**Comment by lbluque** [comment]
Ok, I see the changes in the latest stable version of sklearn.

https://github.com/scikit-learn/scikit-learn/blob/baf0ea25d6dd034403370fea552b21a6776bef18/sklearn/linear_model/_base.py#L214

Feel free to fix by importing `_preprocess_data` from `sklearn.linear_model._base` directly.  But please update sparse-lm requirements to list the correct versions of skearn that are supported.

**Comment by lbluque** [comment]
The minimum sklearn version should be set [here](https://github.com/CederGroupHub/sparse-lm/blob/5b99a0d47aacf0d1611ab107fd8a44cf06a00d76/pyproject.toml#L14)

**Comment by lbluque** [comment]
fixed in #6
